mne_bids/utils.py

```diff
@@ -29,6 +30,142 @@
+    example, electrode location TSV files do not need a task field.
+
+    Parameters
+    ----------
+    subject : str | None
+        The subject ID. Corresponds to "sub".
+    session : str | None
+        The session for a item. Corresponds to "ses".
+    task : str | None
+        The task for a item. Corresponds to "task".
+    acquisition: str | None
+        The acquisition parameters for the item. Corresponds to "acq".
+    run : int | None
+        The run number for this item. Corresponds to "run".
+    processing : str | None
+        The processing label for this item. Corresponds to "proc".
+    recording : str | None
+        The recording name for this item. Corresponds to "recording".
+    space : str | None
+        The coordinate space for an anatomical file. Corresponds to "space".
+    prefix : str | None
+        The prefix for the filename to be created. E.g., a path to the folder
+        in which you wish to create a file with this name.
+    suffix : str | None
+        The suffix for the filename to be created. E.g., 'audio.wav'.
+
+    Returns
+    -------
+    filename : str
+        The BIDS filename you wish to create.
+
+    Examples
+    --------
+    >>> bids_basename = make_bids_basename(subject='test', session='two', task='mytask', suffix='data.csv')  # noqa: E501
+    >>> print(bids_basename)
+    sub-test_ses-two_task-mytask_data.csv
+    >>> print(bids_basename.__repr__())
+    BIDSPath (sub-test_ses-two_task-mytask_data.csv)
+    """
+
+    def __init__(self, sub=None, ses=None, task=None,
+                 acq=None, run=None, proc=None,
+                 rec=None, space=None,
+                 prefix=None, suffix=None):
+        self.sub = sub
+        self.ses = ses
+        self.task = task
+        self.acq = acq
+        self.run = run
+        self.proc = proc
+        self.rec = rec
+        self.space = space
+        self.prefix = prefix
+        self.suffix = suffix
+
+        # run string representation to check validity of arguments
+        self._get_name()
+
+    @property
+    def entities(self):
+        """Return dictionary of the BIDS entities."""
+        keys = ('sub', 'ses', 'task', 'acq',
+                'proc', 'acq', 'run', 'rec',
+                'space', 'suffix', 'prefix')
+        entities = OrderedDict()
+
+        for key in keys:
+            value = getattr(self, key, None)
+            if value is not None:
+                entities[key] = value
+        return entities
+
+    def __setitem__(self, key, value):
+        """Set item as a dictionary, and perform validation checks."""
+        if key not in ('sub', 'ses', 'task', 'acq',
+                       'proc', 'acq', 'run', 'rec',
+                       'space', 'suffix', 'prefix'):
+            raise ValueError('Key must be one of blah, got %s' % key)
+        setattr(self, key, value)
+
+    def __str__(self):
+        """Return the string representation of the path."""
+        return self._get_name()
```

Comment:
why make another function for just 2 lines? This is not good. You are sending the person reading the code on a treasure hunt!